src/librustc/ty/layout.rs

```diff
@@ -1490,27 +1492,34 @@ impl<'gcx, 'tcx, T: HasTyCtxt<'gcx>> HasTyCtxt<'gcx> for LayoutCx<'tcx, T> {
     }
 }
 
-pub trait MaybeResult<T> {
+pub trait MaybeResult<T, E> {
     fn from_ok(x: T) -> Self;
     fn map_same<F: FnOnce(T) -> T>(self, f: F) -> Self;
+    fn to_result(self) -> Result<T, E>;
```

Comment:
@oli-obk I am trying to add this `to_result `. Having a lot of fallout. First time with traits and lifetimes. So I am not able to progress.

This is the latest error I get:
```
error[E0207]: the type parameter `E` is not constrained by the impl trait, self type, or predicates                                                                         
    --> src/librustc/ty/layout.rs:1614:19
     |
1614 | impl<'a, 'tcx, C, E> TyLayoutMethods<'tcx, C> for Ty<'tcx>
     |                   ^ unconstrained type parameter

error: aborting due to previous error
```

Help please!

Reply:
You can make `E` an associated type instead of a generic parameter.